app/mbm/templates/mbm/index.html

```diff
@@ -9,6 +9,24 @@
   <link rel="stylesheet" href="{% static 'css/spin.css' %}" />
 {% endblock %}
 
+{% block head_js %}
+<!-- Matomo -->
+<script>
+  var _paq = window._paq = window._paq || [];
+  /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
+  _paq.push(['trackPageView']);
+  _paq.push(['enableLinkTracking']);
+  (function() {
+    var u="//analytics.jeancochrane.com/";
+    _paq.push(['setTrackerUrl', u+'matomo.php']);
+    _paq.push(['setSiteId', '1']);
+    var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
+    g.async=true; g.src=u+'matomo.js'; s.parentNode.insertBefore(g,s);
```

Comment:
**[Question, non-blocking]** Is there a reason that it's necessary for this setup logic to be wrapped in an immediately-executed function expression rather than defined in the body of the enclosing `<script>`? Similarly, I'm curious why this snippet chooses to generate the `<script async>` tag programmatically instead of just defining it in the HTML body?

Reply:
The IIFE is to avoid namespace collisions in the global scope with those 1-letter variables, which are pretty common in minified scripts. No clue why they generate the script tag at runtime. I tried googling it and the stackoverflow explanations are very unconvincing!